### PR TITLE
Update some internal interfaces

### PR DESCRIPTION
**Pull request by cw9**
Hide some functions/interfaces that are not used outside the package and can change internal state

@xichen2020

### CODE REVIEW

File: placement/algo/sharded_helper.go [hunk 1]
```diff
@@ -67,25 +67,25 @@ type PlacementHelper interface {
 	// HasRackConflict checks if the rack constraint is violated when moving the shard to the target rack.
 	HasRackConflict(shard uint32, from placement.Instance, toRack string) bool
 
-	// PlaceShards distributes shards to the instances in the helper, with aware of where are the shards coming from.
-	PlaceShards(shards []shard.Shard, from placement.Instance, candidates []placement.Instance) error
+	// placeShards distributes shards to the instances in the helper, with aware of where are the shards coming from.
```
Comment (xichen2020) [inline placement/algo/sharded_helper.go]: It's somewhat unconventional to have public interfaces with private APIs. If we don't want users to use these APIs, perhaps we should just remove them or move them to a private interface?

Comment (cw9) [reply placement/algo/sharded_helper.go]: Sounds good, separated the interfaces

File: placement/algo/sharded_helper.go [hunk 2]
```diff
@@ -67,28 +88,11 @@ type PlacementHelper interface {
...
-
-	// ReclaimLeavingShards reclaims all the leaving shards on the given instance
-	// by pulling them back from the rest of the cluster.
-	ReclaimLeavingShards(instance placement.Instance)
-
-	// ReturnInitializingShards returns all the initializing shards on the given instance
-	// by returning them back to the original owners.
-	ReturnInitializingShards(instance placement.Instance)
+	// placeShards distributes shards to the instances in the helper, with aware of where are the shards coming from.
+	placeShards(shards []shard.Shard, from placement.Instance, candidates []placement.Instance) error
```
Comment (xichen2020) [inline placement/algo/sharded_helper.go]: Should this be moved as well?

Comment (cw9) [reply placement/algo/sharded_helper.go]: lol, missed this. Good catch

**Pull request merged**